bioio_ome_zarr/writers/ome_zarr_writer.py

```diff
@@ -444,31 +458,40 @@ def write_timepoints(
             for i in range(self.ndim)
         )
 
+        # Build compute graph across all levels, then compute once.
+        ops = []
+        cur = batch_arr
         for level_index in range(self.num_levels):
-            if level_index == 0:
-                level_block = batch_arr
-            else:
+            if level_index > 0:
                 nextshape = list(self.level_shapes[level_index])
                 nextshape[axis_t] = total_T
-                level_block = resize(
-                    batch_arr,
-                    tuple(nextshape),
-                    order=0,
-                ).astype(batch_arr.dtype)
+                cur = resize(cur, tuple(nextshape), order=0).astype(cur.dtype)
+
+            # Align to target chunks
+            tgt_chunks = self.datasets[level_index].chunks
+            src = cur if cur.chunks == tgt_chunks else cur.rechunk(tgt_chunks)
```

Comment:
Nit: `write_full_volume` and `write_timepoints` are starting to get a bit of repeated code. Maybe they can share?